eZ/Publish/API/Repository/Tests/SearchServiceAggregationTest.php

```diff
@@ -0,0 +1,886 @@
+                'publication_date',
+                [
+                    new RangeAggregationResultEntry(
+                        new Range(null, new DateTime('2003-01-01')),
+                        6
+                    ),
+                    new RangeAggregationResultEntry(
+                        new Range(new DateTime('2003-01-01'), new DateTime('2004-01-01')),
+                        2
+                    ),
+                    new RangeAggregationResultEntry(
+                        new Range(new DateTime('2004-01-01'), null),
+                        10
+                    ),
+                ]
+            ),
+        ];
+
+        yield LanguageTermAggregation::class => $this->createTermAggregationTestCase(
+            new LanguageTermAggregation('language'),
+            [
+                'eng-US' => 16,
+                'eng-GB' => 2,
+            ],
+            [$this->getRepository()->getContentLanguageService(), 'loadLanguage']
+        );
+
+        yield ObjectStateTermAggregation::class => $this->createTermAggregationTestCase(
+            new ObjectStateTermAggregation('object_state', 'ez_lock'),
+            [
+                // TODO: Change the state of some content objects to have better test data
+                'not_locked' => 18,
+            ],
+            function (string $identifier): ObjectState {
+                $objectStateService = $this->getRepository()->getObjectStateService();
+
+                static $objectStateGroup = null;
+                if ($objectStateGroup === null) {
+                    $objectStateGroup = $objectStateService->loadObjectStateGroupByIdentifier('ez_lock');
+                }
+
+                return $objectStateService->loadObjectStateByIdentifier($objectStateGroup, $identifier);
+            }
+        );
+
+        yield SectionTermAggregation::class => $this->createTermAggregationTestCase(
+            new SectionTermAggregation('section'),
+            [
+                'users' => 8,
+                'media' => 4,
+                'standard' => 2,
+                'setup' => 2,
+                'design' => 2,
+            ],
+            [$this->getRepository()->getSectionService(), 'loadSectionByIdentifier']
+        );
+
+        yield UserMetadataTermAggregation::class . '::OWNER' => $this->createTermAggregationTestCase(
+            new UserMetadataTermAggregation('owner', UserMetadataTermAggregation::OWNER),
+            [
+                'admin' => 18,
+            ],
+            [$this->getRepository()->getUserService(), 'loadUserByLogin']
+        );
+
+        yield UserMetadataTermAggregation::class . '::GROUP' => $this->createTermAggregationTestCase(
+            new UserMetadataTermAggregation('user_group', UserMetadataTermAggregation::GROUP),
+            [
+                12 => 18,
+                14 => 18,
+                4 => 18,
+            ],
+            [$this->getRepository()->getUserService(), 'loadUserGroup']
```

Comment:
code repetitions: `$this->getRepository()->getUserService()` should be fetched only once. Ideally all of those should be variables.

Reply:
Fixed in 672a0ea